tests/scripts/restore-mon-quorum-using-osds.sh

```diff
@@ -0,0 +1,318 @@
+
+    TOOL=$(kubectl get -n "${NAMESPACE}" po -l app=rook-ceph-tools -o json | jq -r '.items[0].metadata.name')
+}
+
+RESTORE=0
+while getopts hrf:n:b:t: OPT
+do
+  case "${OPT}" in
+    h)
+        usage 0
+        ;;
+    f)
+        FSID="${OPTARG}"
+        ;;
+    n)
+        NAMESPACE="${OPTARG}"
+        ;;
+    b)
+        BACKUP_DIR="${OPTARG}"
+        ;;
+    t)
+        TMP_DIR="${OPTARG}"
+        ;;
+    r)
+        RESTORE=1
+        ;;
+    *)
+        echo "invalid option: ${OPT}"
+        usage 1
+        ;;
+  esac
+done
+
+shift $((OPTIND - 1))
+
+if [ "${RESTORE}" = 1 ]; then
+  if [ -z "${FSID}" -o -z "${NAMESPACE}" -o -z "${BACKUP_DIR}" ]; then
+    usage 1
+  fi
+  init
+  restore
+  exit 0
+fi
+
+if [ -z "${FSID}" -o -z "${NAMESPACE}" -o -z "${BACKUP_DIR}" -o -z "${TMP_DIR}" ]; then
+    usage 1
+fi
+
+init
+
+echo '[STEP1] taking backup'
+
+for mon in ${MONS}; do
+    kubectl -n "${NAMESPACE}" get deployment "${mon}" -o yaml > "${BACKUP_DIR}"/"${mon}".yaml
+done
+
+for osd in ${OSDS}; do
+    kubectl -n "${NAMESPACE}" get deployment "${osd}" -o yaml > "${BACKUP_DIR}"/"${osd}".yaml
+done
+
+kubectl -n "${NAMESPACE}" get deployment "${TOOL_DEPLOYMENT}" -o yaml > "${BACKUP_DIR}"/"${TOOL_DEPLOYMENT}".yaml
+
+echo '[STEP1] done'
+
+echo '[STEP2] suspending operator pod'
+
+kubectl -n "${NAMESPACE}" scale deployment rook-ceph-operator --replicas=0
+wait_pods rook-ceph-operator 0
+
+echo '[STEP2] done'
+
+echo '[STEP3] suspending mon pods and osd pods'
+
+for mon in ${MONS}; do
+    kubectl -n "${NAMESPACE}" patch deployment "${mon}" -p '{"spec": {"template": {"spec": {"containers": [{"name": "mon", "command": ["sleep", "infinity"], "args": []}]}}}}'
+    kubectl -n "${NAMESPACE}" patch deployment "${mon}" --type=json --patch='[{"op":"remove", "path":"/spec/template/spec/containers/0/livenessProbe"}]'
+done
+
+wait_pods rook-ceph-mon "${DESIRED_MON_COUNT}" Running
+
+for osd in ${OSDS}; do
+    kubectl -n "${NAMESPACE}" patch deployment "${osd}" -p '{"spec": {"template": {"spec": {"containers": [{"name": "osd", "command": ["sleep", "infinity"], "args": []}]}}}}'
+    kubectl -n "${NAMESPACE}" patch deployment "${osd}" --type=json --patch='[{"op":"remove", "path":"/spec/template/spec/containers/0/livenessProbe"}]'
+done
+
+wait_pods rook-ceph-osd "${DESIRED_OSD_COUNT}" Running
+
+echo '[STEP3] done'
+
+echo '[STEP4] injecting mon-keyring to toolbox for the subsequent steps'
+
+kubectl -n "${NAMESPACE}" patch deployment "${TOOL_DEPLOYMENT}" --type=json \
+ --patch='[
+     {"op":"add", "path":"/spec/template/spec/volumes/-", "value": {"name": "mons-keyring", "secret": {"secretName": "rook-ceph-mons-keyring"}}},
+     {"op":"add", "path":"/spec/template/spec/containers/0/volumeMounts/-", "value": {"name": "mons-keyring", "mountPath": "/keyrings"}}
+ ]'
+
+kubectl -n "${NAMESPACE}" delete pod -l app=rook-ceph-tools
+wait_pods rook-ceph-tools 1 Running
```

Comment:
I forgot to say the important limitation. This script seems not to work with new cluster. Guess your cluster is so too. Is my assumption correct?

Reply:
Oh, it's not same as what I imagined. In my new cluster case, this script stuck in enabling msgr2.

> '[' 1 = '       1' ']'

I should fix the way to use shell variables.

Reply:
@travisn Could you show me your cluster.yaml?

Reply:
> I forgot to say the important limitation. This script seems not to work with new cluster. Guess your cluster is so too. Is my assumption correct?

Yes, it was a new minikube cluster. What version of a cluster is it expected to work with currently?

> @travisn Could you show me your cluster.yaml?

@satoru-takeuchi I was using [cluster-test.yaml](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/cluster-test.yaml) except I changed the mon count to 3.

Reply:
> Yes, it was a new minikube cluster

I'll make a procedure to make this script work when adding the integration test for this script in the next PR. I guess running after smoke test is enough.

> What version of a cluster is it expected to work with currently?

I test this script only with Rook v1.6.3 + Ceph 16.2.4 cluster.

Reply:
Ok, not sure what would have changed from master that would affect this not to work since v1.6.3. Anyway, as part of a future PR it would be really nice to run this script as part of an integration test so we will know if something breaks it.